odilo.py

```diff
@@ -0,0 +1,598 @@
+        series_position = book.get('seriesPosition')
+
+        contributors = []
+        author = book.get('author')
+        if author:
+            roles = [Contributor.AUTHOR_ROLE]
+            contributor = ContributorData(sort_name=author, display_name=author, roles=roles, biography=None)
+            contributors.append(contributor)
+
+        publisher = book.get('publisher')
+
+        # Metadata --> Marc21 260$c
+        published = book.get('publicationDate')
+        if not published:
+            # yyyyMMdd --> record creation date
+            published = book.get('releaseDate')
+
+        if published:
+            try:
+                published = datetime.datetime.strptime(published, "%Y%m%d")
+            except ValueError as e:
+                cls.log.warn('Cannot parse publication date from: ' + published + ', message: ' + e.message)
+
+        # yyyyMMdd --> record last modification date
+        last_update = book['modificationDate']
+        if last_update:
+            try:
+                last_update = datetime.datetime.strptime(last_update, "%Y%m%d")
+            except ValueError as e:
+                cls.log.warn('Cannot parse last update date from: ' + last_update + ', message: ' + e.message)
+
+        language = book.get('language')
+
+        subjects = []
+        for subject in book.get('subjects', []):
+            subjects.append(SubjectData(type=Subject.TAG, identifier=subject, weight=100))
+
+        grade_level = book.get('gradeLevel')
+        if grade_level:
+            subject = SubjectData(type=Subject.GRADE_LEVEL, identifier=grade_level, weight=10)
+            subjects.append(subject)
+
+        medium = None
+        formats = []
+        for format_received in book.get('formats', []):
+            # Both formats (PDF and EPUB) behave the same, and are basically the same thing for the final users.
+            if format_received in ('PDF', 'EPUB'):
+                format_received = 'EBOOK_STREAMING'
+            if format_received in cls.format_data_for_odilo_format:
+                content_type, drm_scheme = cls.format_data_for_odilo_format.get(format_received)
+                formats.append(FormatData(content_type, drm_scheme))
+                if not medium:
+                    medium = cls.odilo_medium_to_simplified_medium.get(format_received)
+            else:
+                cls.log.warn('Unrecognized format received: ' + format_received)
+
+        if not medium:
+            medium = Edition.BOOK_MEDIUM
+
+        identifiers = []
+        isbn = book.get('isbn')
+        if isbn:
+            if len(isbn) == 10:
+                isbn = isbnlib.to_isbn13(isbn)
+            identifiers.append(IdentifierData(Identifier.ISBN, isbn, 1))
+
+        # A cover
+        links = []
+        cover_image_url = book.get('coverImageUrl')
+        if cover_image_url:
+            image_data = cls.image_link_to_linkdata(cover_image_url, Hyperlink.IMAGE)
+            if image_data:
```

Comment:
Looking at your example I see that the cover image sent through the API (http://pruebasotk.odilotk.es/public/OdiloPlace_eduDistUS/pg54159_225x318.jpg) is about 200x300 pixels. This is approximately the size we use for thumbnails. There's a larger image at http://pruebasotk.odilotk.es/public/OdiloPlace_eduDistUS/pg54159.jpg but the API doesn't link to it.

The best thing to do here would be to modify the API to send both URLs. This code would then create two `LinkData` objects, one for the full image and one for the thumbnail. The thumbnail's `LinkData` would be set as `.thumbnail` on the full image's `LinkData`.

The full image is used by the metadata wrangler to scale down a cover image to the correct dimensions for the SimplyE app.

Reply:
OK, I have added the field 'originalImageUrl' to the record metadata retrieved from the API, and used it at the integration as the main image and the other 'coverImageUrl' as the thumbnail.
Note that "the bigger" image you said is actually the "original image" uploaded when cataloging the resource, which doesn't mean that it's always going to be bigger, although it is the most usual.